debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -795,16 +798,48 @@ public OffsetStorageReader offsetStorageReader() {
                                     // this thread is still set as the running thread -> we were not interrupted
                                     // due the stop() call -> probably someone else called the interrupt on us ->
                                     // -> we should raise the interrupt flag
+
                                     Thread.currentThread().interrupt();
                                 }
                                 break;
                             }
-                            catch (RetriableException e) {
-                                LOGGER.info("Retrieable exception thrown, connector will be restarted", e);
+                            catch (RetriableException retriableException) {
+                                LOGGER.info("Retriable exception thrown, connector will be restarted", retriableException);
                                 // Retriable exception should be ignored by the engine
                                 // and no change records delivered.
                                 // The retry is handled in io.debezium.connector.common.BaseSourceTask.poll()
                             }
+                            catch (ConnectException connectException) {
+                                // In case of ConnectException, retry starting the connector with a backoff strategy
+                                int maxRetries = config.getInteger(CommonConnectorConfig.ERRORS_MAX_RETRIES);
```

Comment:
There should be a dedicated method `CommonConnectorConfig::getMaxRetriesOnError()`.